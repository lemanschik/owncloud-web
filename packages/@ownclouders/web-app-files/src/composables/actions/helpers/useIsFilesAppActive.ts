import { computed, unref } from 'vue'
import { activeApp, useRoute } from 'web-pkg/src/composables'

const isFilesAppActive = (activeApp: string): boolean => {
  // FIXME: we should use this constant but it somehow breaks the unit tests
  // return activeApp === FilesApp.appInfo.id
  return activeApp === 'files'
}

export const useIsFilesAppActive = () => {
  const currentRoute = useRoute()

  return computed(() => isFilesAppActive(activeApp(unref(currentRoute))))
}
