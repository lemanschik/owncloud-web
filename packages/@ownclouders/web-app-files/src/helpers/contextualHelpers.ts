import { ConfigurationManager } from 'web-pkg'
import { omit } from 'lodash-es'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

export interface ContextualHelperDataListItem {
  text: string
  headline?: boolean
}
export interface ContextualHelperData {
  title: string
  text?: string
  list?: ContextualHelperDataListItem[]
  readMoreLink?: string
}

export interface ContextualHelperOptions {
  configurationManager: ConfigurationManager
}

export const shareInviteCollaboratorHelp = (options: ContextualHelperOptions) =>
  filterContextHelper(
    {
      title: $gettext('Share with people'),
      text: $gettext(
        'Use the input field to search for users and groups. Select them to share the item.'
      ),
      list: [
        { text: $gettext('Subfolders'), headline: true },
        {
          text: $gettext(
            'If you share a folder, all of its contents and subfolders will be shared as well.'
          )
        },
        { text: $gettext('Notification'), headline: true },
        {
          text: $gettext(
            'People you share resources with will be notified via email or in-app notification.'
          )
        },
        { text: $gettext('Incognito'), headline: true },
        {
          text: $gettext('People you share resources with can not see who else has access.')
        },
        { text: $gettext('“via folder”'), headline: true },
        {
          text: $gettext(
            'The “via folder” is shown next to a share, if access has already been given via a parent folder. Click on the “via folder” to edit the share on its parent folder.'
          )
        }
      ],
      readMoreLink: 'https://doc.owncloud.com/webui/next/owncloud_web/web_for_users.html#sharing'
    },
    options
  )

export const shareInviteCollaboratorHelpCern = (options: ContextualHelperOptions) =>
  filterContextHelper(
    {
      title: '',
      list: [
        { text: $gettext('Search for service or secondary Account'), headline: true },
        {
          text: $gettext(
            'To search for service or secondary accounts prefix the username with "a:" (like "a:doe") and for guest accounts prefix the username with "l:" (like "l:doe").'
          )
        }
      ]
    },
    options
  )

export const shareSpaceAddMemberHelp = (options: ContextualHelperOptions) =>
  filterContextHelper(
    {
      title: $gettext('Add members to this Space'),
      text: $gettext('Enter a name to add people or groups as members to this Space.'),
      list: [
        { text: $gettext('Member capabilities'), headline: true },
        {
          text: $gettext(
            'Members are able to see who has access to this space and access all files in this space. Read or write permissions can be set by assigning a role.'
          )
        },
        { text: $gettext('Space manager capabilities'), headline: true },
        {
          text: $gettext(
            'Members with the Manager role are able to edit all properties and content of a Space, such as adding or removing members, sharing subfolders with non-members, or creating links to share.'
          )
        }
      ],
      readMoreLink: 'https://doc.owncloud.com/webui/next/owncloud_web/web_for_users.html#sharing'
    },
    options
  )
export const shareViaLinkHelp = (options: ContextualHelperOptions) =>
  filterContextHelper(
    {
      title: $gettext('Choose how access is granted'),
      list: [
        {
          text: $gettext('Only invited people can access'),
          headline: true
        },
        {
          text: $gettext(
            'Account and login is required. Only people from the list "Invited people" can access.'
          )
        },
        {
          text: $gettext('Everyone with the link'),
          headline: true
        },
        {
          text: $gettext(
            'No login required. Everyone with the link can access. If you share this link with people from the list "Invited people", they need to login so that their individual assigned permissions can take effect. If they are not logged-in, the permissions of the link take effect.'
          )
        }
      ],
      readMoreLink: 'https://doc.owncloud.com/webui/next/owncloud_web/web_for_users.html#sharing'
    },
    options
  )
export const shareViaIndirectLinkHelp = (options: ContextualHelperOptions) =>
  filterContextHelper(
    {
      title: $gettext('What are indirect links?'),
      text: $gettext('Indirect links are links giving access by a parent folder.'),
      list: [
        {
          text: $gettext('How to edit indirect links'),
          headline: true
        },
        {
          text: $gettext(
            'Indirect links can only be edited in their parent folder. Click on the folder icon below the link to navgate to the parent folder.'
          )
        }
      ],
      readMoreLink: 'https://doc.owncloud.com/webui/next/owncloud_web/web_for_users.html#sharing'
    },
    options
  )

const filterContextHelper = (
  data: ContextualHelperData,
  options?: ContextualHelperOptions
): ContextualHelperData => {
  if (options.configurationManager.options.contextHelpersReadMore === false) {
    return omit(data, 'readMoreLink')
  }
  return data
}
