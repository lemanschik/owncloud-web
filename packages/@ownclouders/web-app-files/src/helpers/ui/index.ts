export * as fileList from './filesList'
export * as resourceTable from './resourceTable'
export * as resourceTiles from './resourceTiles'
