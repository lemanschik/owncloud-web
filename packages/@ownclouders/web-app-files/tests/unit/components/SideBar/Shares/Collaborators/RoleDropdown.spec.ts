import RoleDropdown from 'web-app-files/src/components/SideBar/Shares/Collaborators/RoleDropdown.vue'
import { PeopleShareRoles, SharePermissions } from 'web-client/src/helpers/share'
import {
  createStore,
  defaultPlugins,
  mount,
  shallowMount,
  defaultStoreMockOptions
} from 'web-test-helpers'
import { mock } from 'jest-mock-extended'
import { Resource } from 'web-client'

const selectors = {
  recipientRoleBtn: '.files-recipient-role-select-btn',
  roleButton: '.files-recipient-role-drop-btn',
  roleButtonPrefix: '#files-recipient-role-drop-btn-',
  collaboratorsPermissionPrefix: '#files-collaborators-permission-',
  customPermissionsDrop: '.files-recipient-custom-permissions-drop',
  cancelCustomPermissionsBtn: '.files-recipient-custom-permissions-drop-cancel',
  confirmCustomPermissionsBtn: '.files-recipient-custom-permissions-drop-confirm'
}

describe('RoleDropdown', () => {
  it.each(['folder', 'file'])(
    'renders a button with invite text if no existing role given for resource type %s',
    (type) => {
      const { wrapper } = getWrapper({ mountType: shallowMount, resourceType: type })
      expect(wrapper.html()).toMatchSnapshot()
    }
  )
  it.each(['folder', 'file'])(
    'renders a button with existing role if given for resource type %s',
    (type) => {
      const { wrapper } = getWrapper({
        mountType: shallowMount,
        existingRole: PeopleShareRoles.list(type === 'folder')[0]
      })
      expect(wrapper.html()).toMatchSnapshot()
    }
  )
  it('does not render a button if only one role is available', () => {
    const { wrapper } = getWrapper({
      mountType: shallowMount,
      canShare: true,
      incomingParentShare: {}
    })
    expect(wrapper.find(selectors.recipientRoleBtn).exists()).toBeFalsy()
  })
  it('emits "optionChange"-event on role click', async () => {
    const { wrapper } = getWrapper()
    ;(wrapper.vm.$refs.rolesDrop as any).tippy = { hide: jest.fn() }
    await wrapper.find(selectors.roleButton).trigger('click')
    expect(wrapper.emitted('optionChange')).toBeTruthy()
  })
  describe('custom permissions', () => {
    it.each(['folder', 'file'])(
      'opens custom permissions drop when custom permissions item in the roles gets selected for resource type %s',
      async (type) => {
        const isFolder = type === 'folder'
        const { wrapper } = getWrapper({ resourceType: type })
        const customPermissionsDrop = wrapper.findComponent<any>(selectors.customPermissionsDrop)
        const showHideMock = jest.fn()
        ;(customPermissionsDrop.vm as any).show = showHideMock
        ;(wrapper.vm.$refs.rolesDrop as any).tippy = { hide: showHideMock }

        await wrapper
          .find(selectors.roleButtonPrefix + PeopleShareRoles.custom(isFolder).name)
          .trigger('click')

        expect(showHideMock).toHaveBeenCalled()
      }
    )
    it('can be cancelled and restored', async () => {
      const { wrapper } = getWrapper()
      const showRoleDropMock = jest.fn()
      const hideCustomPermissionsDropMock = jest.fn()
      ;(wrapper.vm.$refs.rolesDrop as any).show = showRoleDropMock
      ;(wrapper.vm.$refs.customPermissionsDrop as any).hide = hideCustomPermissionsDropMock
      await wrapper.find(selectors.cancelCustomPermissionsBtn).trigger('click')

      expect(showRoleDropMock).toHaveBeenCalled()
      expect(hideCustomPermissionsDropMock).toHaveBeenCalled()
      expect(wrapper.vm.customPermissions).toEqual(wrapper.vm.defaultCustomPermissions)
    })
    it('can be confirmed', async () => {
      const { wrapper } = getWrapper()
      const hideCustomPermissionsDropMock = jest.fn()
      ;(wrapper.vm.$refs.customPermissionsDrop as any).hide = hideCustomPermissionsDropMock
      await wrapper.find(selectors.confirmCustomPermissionsBtn).trigger('click')

      expect(hideCustomPermissionsDropMock).toHaveBeenCalled()
      expect(wrapper.emitted('optionChange')).toBeTruthy()
    })
    it.each([
      { permissions: [SharePermissions.read, SharePermissions.share] },
      { permissions: [SharePermissions.read, SharePermissions.share, SharePermissions.delete] },
      {
        permissions: [
          SharePermissions.read,
          SharePermissions.update,
          SharePermissions.share,
          SharePermissions.delete
        ]
      }
    ])("inherits the parents share's permissions: %s", ({ permissions }) => {
      const { wrapper } = getWrapper({
        mountType: shallowMount,
        canShare: true,
        incomingParentShare: { permissions: permissions.reduce((a, b) => a + b.bit, 0) }
      })

      for (const permission of permissions) {
        expect(
          wrapper.find(`${selectors.collaboratorsPermissionPrefix}${permission.key}`).exists()
        ).toBeTruthy()
      }
    })
  })
})

function getWrapper({
  mountType = mount,
  existingRole = null,
  resourceType = 'folder',
  canShare = false,
  incomingParentShare = null
} = {}) {
  const store = createStore(defaultStoreMockOptions)
  return {
    wrapper: mountType(RoleDropdown, {
      props: {
        existingRole,
        allowSharePermission: true
      },
      global: {
        plugins: [...defaultPlugins(), store],
        renderStubDefaultSlot: true,
        provide: {
          resource: mock<Resource>({
            name: resourceType === 'folder' ? 'testfolder' : 'testfile',
            extension: resourceType === 'folder' ? '' : 'jpg',
            type: resourceType,
            isFolder: resourceType === 'folder',
            canShare: () => canShare
          }),
          incomingParentShare
        }
      }
    })
  }
}
