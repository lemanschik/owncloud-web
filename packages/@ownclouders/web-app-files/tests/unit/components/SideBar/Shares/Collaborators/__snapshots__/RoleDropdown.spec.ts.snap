// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoleDropdown renders a button with existing role if given for resource type file 1`] = `
<span class="oc-flex oc-flex-middle">
  <oc-button-stub appearance="raw" arialabel="Select permission" class="files-recipient-role-select-btn" disabled="false" gapsize="none" id="files-collaborators-role-button-new" justifycontent="center" size="medium" submit="button" type="button" variation="passive">
    <span class="oc-text-truncate">Can view</span>
    <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="arrow-down-s" size="medium" type="span" variation="passive"></oc-icon-stub>
  </oc-button-stub>
  <oc-drop-stub class="files-recipient-role-drop" closeonclick="true" dropid="oc-drop-7" isnested="false" mode="click" offset="0" paddingsize="small" position="bottom-start" toggle="#files-collaborators-role-button-new">
    <oc-list-stub aria-label="Select role for the invitation" class="files-recipient-role-drop-list" raw="false">
      <li>
        <oc-button-stub appearance="raw-inverse" class="files-recipient-role-drop-btn oc-p-s oc-background-primary-gradient selected" disabled="false" gapsize="medium" id="files-recipient-role-drop-btn-viewer" justifycontent="space-between" size="medium" submit="button" type="button" variation="primary">
          <span class="oc-flex oc-flex-middle">
            <oc-icon-stub accessiblelabel="" class="oc-pl-s oc-pr-m" color="" filltype="fill" name="eye" size="medium" type="span" variation="inherit"></oc-icon-stub>
            <role-item-stub allowsharepermission="true" role="[object Object]"></role-item-stub>
          </span>
          <span class="oc-flex">
            <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="check" size="medium" type="span" variation="inherit"></oc-icon-stub>
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" class="files-recipient-role-drop-btn oc-p-s" disabled="false" gapsize="medium" id="files-recipient-role-drop-btn-editor" justifycontent="space-between" size="medium" submit="button" type="button" variation="passive">
          <span class="oc-flex oc-flex-middle">
            <oc-icon-stub accessiblelabel="" class="oc-pl-s oc-pr-m" color="" filltype="fill" name="pencil" size="medium" type="span" variation="inherit"></oc-icon-stub>
            <role-item-stub allowsharepermission="true" role="[object Object]"></role-item-stub>
          </span>
          <span class="oc-flex">
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" class="files-recipient-role-drop-btn oc-p-s" disabled="false" gapsize="medium" id="files-recipient-role-drop-btn-custom" justifycontent="space-between" size="medium" submit="button" type="button" variation="passive">
          <span class="oc-flex oc-flex-middle">
            <oc-icon-stub accessiblelabel="" class="oc-pl-s oc-pr-m" color="" filltype="fill" name="settings-3" size="medium" type="span" variation="inherit"></oc-icon-stub>
            <role-item-stub allowsharepermission="true" role="[object Object]"></role-item-stub>
          </span>
          <span class="oc-flex">
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
  </oc-drop-stub>
  <oc-drop-stub class="files-recipient-custom-permissions-drop" closeonclick="false" dropid="oc-drop-8" isnested="false" mode="manual" paddingsize="remove" position="bottom-start" target="#files-collaborators-role-button-new" toggle="">
    <h4 class="oc-text-bold oc-m-rm oc-px-m oc-pt-m oc-pb-s">Custom permissions</h4>
    <oc-list-stub raw="false">
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="true" hidelabel="false" id="files-collaborators-permission-read" label="View" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-update" label="Update" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-create" label="Create" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-delete" label="Delete" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-share" label="Share" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
    </oc-list-stub>
    <div class="files-recipient-custom-permissions-drop-cancel-confirm-btns oc-px-m oc-py-s oc-mt-m oc-rounded-bottom">
      <oc-button-stub appearance="outline" class="files-recipient-custom-permissions-drop-cancel" disabled="false" gapsize="medium" justifycontent="center" size="small" submit="button" type="button" variation="passive">Cancel</oc-button-stub>
      <oc-button-stub appearance="filled" class="files-recipient-custom-permissions-drop-confirm oc-ml-s" disabled="false" gapsize="medium" justifycontent="center" size="small" submit="button" type="button" variation="primary">Apply</oc-button-stub>
    </div>
  </oc-drop-stub>
</span>
`;

exports[`RoleDropdown renders a button with existing role if given for resource type folder 1`] = `
<span class="oc-flex oc-flex-middle">
  <oc-button-stub appearance="raw" arialabel="Select permission" class="files-recipient-role-select-btn" disabled="false" gapsize="none" id="files-collaborators-role-button-new" justifycontent="center" size="medium" submit="button" type="button" variation="passive">
    <span class="oc-text-truncate">Can view</span>
    <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="arrow-down-s" size="medium" type="span" variation="passive"></oc-icon-stub>
  </oc-button-stub>
  <oc-drop-stub class="files-recipient-role-drop" closeonclick="true" dropid="oc-drop-5" isnested="false" mode="click" offset="0" paddingsize="small" position="bottom-start" toggle="#files-collaborators-role-button-new">
    <oc-list-stub aria-label="Select role for the invitation" class="files-recipient-role-drop-list" raw="false">
      <li>
        <oc-button-stub appearance="raw-inverse" class="files-recipient-role-drop-btn oc-p-s oc-background-primary-gradient selected" disabled="false" gapsize="medium" id="files-recipient-role-drop-btn-viewer" justifycontent="space-between" size="medium" submit="button" type="button" variation="primary">
          <span class="oc-flex oc-flex-middle">
            <oc-icon-stub accessiblelabel="" class="oc-pl-s oc-pr-m" color="" filltype="fill" name="eye" size="medium" type="span" variation="inherit"></oc-icon-stub>
            <role-item-stub allowsharepermission="true" role="[object Object]"></role-item-stub>
          </span>
          <span class="oc-flex">
            <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="check" size="medium" type="span" variation="inherit"></oc-icon-stub>
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" class="files-recipient-role-drop-btn oc-p-s" disabled="false" gapsize="medium" id="files-recipient-role-drop-btn-editor" justifycontent="space-between" size="medium" submit="button" type="button" variation="passive">
          <span class="oc-flex oc-flex-middle">
            <oc-icon-stub accessiblelabel="" class="oc-pl-s oc-pr-m" color="" filltype="fill" name="pencil" size="medium" type="span" variation="inherit"></oc-icon-stub>
            <role-item-stub allowsharepermission="true" role="[object Object]"></role-item-stub>
          </span>
          <span class="oc-flex">
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" class="files-recipient-role-drop-btn oc-p-s" disabled="false" gapsize="medium" id="files-recipient-role-drop-btn-custom" justifycontent="space-between" size="medium" submit="button" type="button" variation="passive">
          <span class="oc-flex oc-flex-middle">
            <oc-icon-stub accessiblelabel="" class="oc-pl-s oc-pr-m" color="" filltype="fill" name="settings-3" size="medium" type="span" variation="inherit"></oc-icon-stub>
            <role-item-stub allowsharepermission="true" role="[object Object]"></role-item-stub>
          </span>
          <span class="oc-flex">
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
  </oc-drop-stub>
  <oc-drop-stub class="files-recipient-custom-permissions-drop" closeonclick="false" dropid="oc-drop-6" isnested="false" mode="manual" paddingsize="remove" position="bottom-start" target="#files-collaborators-role-button-new" toggle="">
    <h4 class="oc-text-bold oc-m-rm oc-px-m oc-pt-m oc-pb-s">Custom permissions</h4>
    <oc-list-stub raw="false">
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="true" hidelabel="false" id="files-collaborators-permission-read" label="View" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-update" label="Update" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-create" label="Create" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-delete" label="Delete" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-share" label="Share" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
    </oc-list-stub>
    <div class="files-recipient-custom-permissions-drop-cancel-confirm-btns oc-px-m oc-py-s oc-mt-m oc-rounded-bottom">
      <oc-button-stub appearance="outline" class="files-recipient-custom-permissions-drop-cancel" disabled="false" gapsize="medium" justifycontent="center" size="small" submit="button" type="button" variation="passive">Cancel</oc-button-stub>
      <oc-button-stub appearance="filled" class="files-recipient-custom-permissions-drop-confirm oc-ml-s" disabled="false" gapsize="medium" justifycontent="center" size="small" submit="button" type="button" variation="primary">Apply</oc-button-stub>
    </div>
  </oc-drop-stub>
</span>
`;

exports[`RoleDropdown renders a button with invite text if no existing role given for resource type file 1`] = `
<span class="oc-flex oc-flex-middle">
  <oc-button-stub appearance="raw" arialabel="Select permission" class="files-recipient-role-select-btn" disabled="false" gapsize="none" id="files-collaborators-role-button-new" justifycontent="center" size="medium" submit="button" type="button" variation="passive">
    <span class="oc-text-truncate">Can view</span>
    <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="arrow-down-s" size="medium" type="span" variation="passive"></oc-icon-stub>
  </oc-button-stub>
  <oc-drop-stub class="files-recipient-role-drop" closeonclick="true" dropid="oc-drop-3" isnested="false" mode="click" offset="0" paddingsize="small" position="bottom-start" toggle="#files-collaborators-role-button-new">
    <oc-list-stub aria-label="Select role for the invitation" class="files-recipient-role-drop-list" raw="false">
      <li>
        <oc-button-stub appearance="raw-inverse" class="files-recipient-role-drop-btn oc-p-s oc-background-primary-gradient selected" disabled="false" gapsize="medium" id="files-recipient-role-drop-btn-viewer" justifycontent="space-between" size="medium" submit="button" type="button" variation="primary">
          <span class="oc-flex oc-flex-middle">
            <oc-icon-stub accessiblelabel="" class="oc-pl-s oc-pr-m" color="" filltype="fill" name="eye" size="medium" type="span" variation="inherit"></oc-icon-stub>
            <role-item-stub allowsharepermission="true" role="[object Object]"></role-item-stub>
          </span>
          <span class="oc-flex">
            <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="check" size="medium" type="span" variation="inherit"></oc-icon-stub>
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" class="files-recipient-role-drop-btn oc-p-s" disabled="false" gapsize="medium" id="files-recipient-role-drop-btn-editor" justifycontent="space-between" size="medium" submit="button" type="button" variation="passive">
          <span class="oc-flex oc-flex-middle">
            <oc-icon-stub accessiblelabel="" class="oc-pl-s oc-pr-m" color="" filltype="fill" name="pencil" size="medium" type="span" variation="inherit"></oc-icon-stub>
            <role-item-stub allowsharepermission="true" role="[object Object]"></role-item-stub>
          </span>
          <span class="oc-flex">
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" class="files-recipient-role-drop-btn oc-p-s" disabled="false" gapsize="medium" id="files-recipient-role-drop-btn-custom" justifycontent="space-between" size="medium" submit="button" type="button" variation="passive">
          <span class="oc-flex oc-flex-middle">
            <oc-icon-stub accessiblelabel="" class="oc-pl-s oc-pr-m" color="" filltype="fill" name="settings-3" size="medium" type="span" variation="inherit"></oc-icon-stub>
            <role-item-stub allowsharepermission="true" role="[object Object]"></role-item-stub>
          </span>
          <span class="oc-flex">
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
  </oc-drop-stub>
  <oc-drop-stub class="files-recipient-custom-permissions-drop" closeonclick="false" dropid="oc-drop-4" isnested="false" mode="manual" paddingsize="remove" position="bottom-start" target="#files-collaborators-role-button-new" toggle="">
    <h4 class="oc-text-bold oc-m-rm oc-px-m oc-pt-m oc-pb-s">Custom permissions</h4>
    <oc-list-stub raw="false">
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="true" hidelabel="false" id="files-collaborators-permission-read" label="View" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-update" label="Update" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-share" label="Share" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
    </oc-list-stub>
    <div class="files-recipient-custom-permissions-drop-cancel-confirm-btns oc-px-m oc-py-s oc-mt-m oc-rounded-bottom">
      <oc-button-stub appearance="outline" class="files-recipient-custom-permissions-drop-cancel" disabled="false" gapsize="medium" justifycontent="center" size="small" submit="button" type="button" variation="passive">Cancel</oc-button-stub>
      <oc-button-stub appearance="filled" class="files-recipient-custom-permissions-drop-confirm oc-ml-s" disabled="false" gapsize="medium" justifycontent="center" size="small" submit="button" type="button" variation="primary">Apply</oc-button-stub>
    </div>
  </oc-drop-stub>
</span>
`;

exports[`RoleDropdown renders a button with invite text if no existing role given for resource type folder 1`] = `
<span class="oc-flex oc-flex-middle">
  <oc-button-stub appearance="raw" arialabel="Select permission" class="files-recipient-role-select-btn" disabled="false" gapsize="none" id="files-collaborators-role-button-new" justifycontent="center" size="medium" submit="button" type="button" variation="passive">
    <span class="oc-text-truncate">Can view</span>
    <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="arrow-down-s" size="medium" type="span" variation="passive"></oc-icon-stub>
  </oc-button-stub>
  <oc-drop-stub class="files-recipient-role-drop" closeonclick="true" dropid="oc-drop-1" isnested="false" mode="click" offset="0" paddingsize="small" position="bottom-start" toggle="#files-collaborators-role-button-new">
    <oc-list-stub aria-label="Select role for the invitation" class="files-recipient-role-drop-list" raw="false">
      <li>
        <oc-button-stub appearance="raw-inverse" class="files-recipient-role-drop-btn oc-p-s oc-background-primary-gradient selected" disabled="false" gapsize="medium" id="files-recipient-role-drop-btn-viewer" justifycontent="space-between" size="medium" submit="button" type="button" variation="primary">
          <span class="oc-flex oc-flex-middle">
            <oc-icon-stub accessiblelabel="" class="oc-pl-s oc-pr-m" color="" filltype="fill" name="eye" size="medium" type="span" variation="inherit"></oc-icon-stub>
            <role-item-stub allowsharepermission="true" role="[object Object]"></role-item-stub>
          </span>
          <span class="oc-flex">
            <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="check" size="medium" type="span" variation="inherit"></oc-icon-stub>
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" class="files-recipient-role-drop-btn oc-p-s" disabled="false" gapsize="medium" id="files-recipient-role-drop-btn-editor" justifycontent="space-between" size="medium" submit="button" type="button" variation="passive">
          <span class="oc-flex oc-flex-middle">
            <oc-icon-stub accessiblelabel="" class="oc-pl-s oc-pr-m" color="" filltype="fill" name="pencil" size="medium" type="span" variation="inherit"></oc-icon-stub>
            <role-item-stub allowsharepermission="true" role="[object Object]"></role-item-stub>
          </span>
          <span class="oc-flex">
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" class="files-recipient-role-drop-btn oc-p-s" disabled="false" gapsize="medium" id="files-recipient-role-drop-btn-custom" justifycontent="space-between" size="medium" submit="button" type="button" variation="passive">
          <span class="oc-flex oc-flex-middle">
            <oc-icon-stub accessiblelabel="" class="oc-pl-s oc-pr-m" color="" filltype="fill" name="settings-3" size="medium" type="span" variation="inherit"></oc-icon-stub>
            <role-item-stub allowsharepermission="true" role="[object Object]"></role-item-stub>
          </span>
          <span class="oc-flex">
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
  </oc-drop-stub>
  <oc-drop-stub class="files-recipient-custom-permissions-drop" closeonclick="false" dropid="oc-drop-2" isnested="false" mode="manual" paddingsize="remove" position="bottom-start" target="#files-collaborators-role-button-new" toggle="">
    <h4 class="oc-text-bold oc-m-rm oc-px-m oc-pt-m oc-pb-s">Custom permissions</h4>
    <oc-list-stub raw="false">
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="true" hidelabel="false" id="files-collaborators-permission-read" label="View" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-update" label="Update" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-create" label="Create" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-delete" label="Delete" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
      <li class="oc-my-s oc-px-m">
        <oc-checkbox-stub class="oc-mr-xs files-collaborators-permission-checkbox" disabled="false" hidelabel="false" id="files-collaborators-permission-share" label="Share" modelvalue="[object Object],[object Object]" option="[object Object]" outline="false" size="large"></oc-checkbox-stub>
      </li>
    </oc-list-stub>
    <div class="files-recipient-custom-permissions-drop-cancel-confirm-btns oc-px-m oc-py-s oc-mt-m oc-rounded-bottom">
      <oc-button-stub appearance="outline" class="files-recipient-custom-permissions-drop-cancel" disabled="false" gapsize="medium" justifycontent="center" size="small" submit="button" type="button" variation="passive">Cancel</oc-button-stub>
      <oc-button-stub appearance="filled" class="files-recipient-custom-permissions-drop-confirm oc-ml-s" disabled="false" gapsize="medium" justifycontent="center" size="small" submit="button" type="button" variation="primary">Apply</oc-button-stub>
    </div>
  </oc-drop-stub>
</span>
`;
